test/cache_invalidation_test.rb

```diff
@@ -10,6 +10,7 @@ def setup
     @record.save
     @record.reload
     @baz, @bar = @record.associated_records[0], @record.associated_records[1]
+    @record.reload
```

Comment:
The above line loads the association, so `fetch_associated_records` on test_reload_invalidate_cached_objects wasn't actually loading the association from the cache and failed on the assertion for the value in the instance variable that holds the cached association.